src/nvim/terminal.c

```diff
@@ -313,6 +313,10 @@ void terminal_resize(Terminal *term, uint16_t width, uint16_t height)
     return;
   }
 
+  if (height == 0 || width == 0) {
+    return;
+  }
```

Comment:
Above the FOR_ALL_WINDOWS_IN_TAB this wouldn't work yet. If 0 was passed as parameter, the width or height would be set to the current value (not 0, so no return), and then we'd find the minimum (again 0, causing the same problem). 
Also, as you mentioned in a comment to PR #2683, having width & height as a parameter doesn't really add value, since we need to walk across all windows anyway. There are some other problems with this function as well - currently the terminal's size can only decrease, never increase. All this and more is addressed in PR #2683, which can be completed when this PR gets merged in.

The current PR is meant as a minimal bugfix that would enable me to make greater improvements in the other one.